oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/server/ServerBearerTokenAuthenticationConverter.java

```diff
@@ -90,8 +91,20 @@ public void setAllowUriQueryParameter(boolean allowUriQueryParameter) {
 		this.allowUriQueryParameter = allowUriQueryParameter;
 	}
 
-	private static String resolveFromAuthorizationHeader(HttpHeaders headers) {
-		String authorization = headers.getFirst(HttpHeaders.AUTHORIZATION);
+	/**
+	 * Set this value to configure what header is checked when resolving a Bearer Token.
+	 * This value is defaulted to {@link HttpHeaders#AUTHORIZATION}.
+	 *
+	 * This allows other headers to be used such as {@link HttpHeaders#PROXY_AUTHORIZATION}
+	 *
+	 * @param authorizationHeader the header to check when retrieving the Bearer Token.
+	 */
+	public void setAuthorizationHeader(String authorizationHeader) {
+		this.authorizationHeader = authorizationHeader;
+	}
+
+	private static String resolveFromAuthorizationHeader(HttpHeaders headers, String authorizationHeader) {
+		String authorization = headers.getFirst(authorizationHeader);
```

Comment:
Is there a reason `authorizationHeader` needs to be passed in? It seems like it would be a smaller change to use `this.`.

Reply:
I chose to add a parameter since the method is static. I could not access the ```this.authorizationHeader```.

I don't see a reason it needed to be static. See last push.